walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,64 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.Iterator;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    
+    Set<String> attendees = new HashSet<String>(request.getAttendees());
+
+    Map<Integer, Integer> timeSlot = new TreeMap<Integer, Integer>();
+
+    for (Event event : events) {
+      TimeRange when = event.getWhen();
+
+      boolean hasRelatedPerson = false;
+      for (String attendee : event.getAttendees())
+        if (attendees.contains(attendee))
+          hasRelatedPerson = true;
+      
+      if (!hasRelatedPerson)
+        continue;
+
+      if (timeSlot.get(when.start()) == null) 
```

Comment:
Could you help add comments and explains what the timeSlot does and how it stored the time info?

Reply:
OK, I will write it in the code soon.

Reply:
Added!

Reply:
Thanks for the well-explained comments!